src/lib/components/Post.svelte

```diff
@@ -99,18 +129,65 @@
+                    </Button>
+                </DropdownMenu.Trigger>
+                <DropdownMenu.Content class="w-56">
+                    <DropdownMenu.Group>
+                        <DropdownMenu.Item>Share</DropdownMenu.Item>
+                        <DropdownMenu.Separator />
+                        <Dialog.Trigger>Report</Dialog.Trigger>
+                        <Dialog.Content class="sm:max-w-[425px]">
+                            <Dialog.Header>
+                                <Dialog.Title>Report Form</Dialog.Title>
+                                <Dialog.Description>This action cannot be undone</Dialog.Description
+                                >
+                            </Dialog.Header>
+                            <div class="grid gap-2 py-2">
+                                <Select.Root bind:selected={reportData.reason} portal={null}>
+                                    <Select.Trigger class="w-[300px]">
+                                        <Select.Value
+                                            placeholder="What is the reason for the report?"
+                                        />
+                                    </Select.Trigger>
+                                    <Select.Content>
+                                        <Select.Group>
+                                            <Select.Label></Select.Label>
+                                            <Select.Item value="community"
+                                                >Post violates community guidlines.</Select.Item
+                                            >
+                                            <Select.Item value="site"
+                                                >Post violates site guidelines.</Select.Item
+                                            >
+                                        </Select.Group>
+                                    </Select.Content>
+                                </Select.Root>
+                            </div>
+                            <div class="grid gap-2 py-2">
+                                <Label for="Report details" class="text-left">Report Details</Label>
+                                <Input
+                                    id="Report details"
+                                    name="details"
+                                    bind:value={reportData.details}
+                                    class="col-span-3"
+                                />
+                            </div>
+                            <Dialog.Footer>
+                                <button class="Submit Report" on:click={submitReport}>
+                                    Submit Report
+                                </button>
+                            </Dialog.Footer>
+                        </Dialog.Content>
+                    </DropdownMenu.Group>
+                </DropdownMenu.Content>
+            </DropdownMenu.Root>
+        </Dialog.Root>
```

Comment:
The "Report" button in the dropdown looks funky.  This is due to the Dialog being in the DropdownMenu.  Move the Dialog above the card and use the button in the dropdown to set `open = true` to show the report button properly.